### PR TITLE
Vehicles cleanup and fixes

### LINKED ISSUE #1792: Cant load artifacts onto dropships
#### Description of issue
You cant load large artifacts onto dropships.

#### Difference between expected and actual behavior
I expect to be able to load the artifact into the dropship, but you cant.

#### Steps to reproduce 
Try to load a artifact into a dropship.
Fail.

#### Specific information for locating
idk

#### Length of time in which bug has been known to occur
About a few months, whenever the artifacts were added.

#### Client version, Server revision & Game ID

59c3f12b2e6f84315a172906e1795adc08b6399d - alpha - 2019-10-07
b3D-be5P

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of the issue being looked at sooner.

- [ x] Issue could be reproduced at least once
- [ x] Issue could be reproduced by different players
- [x ] Issue could be reproduced in multiple rounds
- [x ] Issue happened in a recent (less than 7 days ago) round
- [ x] [Couldn't find an existing issue about this](https://github.com/HaloSpaceStation/HaloSpaceStation13/issues)

### PR DESCRIPTION
**Pull request by caelaislinn**
A partial rewrite to simplify the code and make it easier to update and maintain. Also fixes #1792

I'm putting the smooth move changelog here because I didn't do it properly in #1836 

:cl: Cael Aislinn
rscadd: As of October 2019 spaceships use a Newtonian flight model. Apply thrust to move in a direction, and alter your heading any time.
bugfix: The Covenant and UNSC can now load Forerunner artifacts onto their dropships (machinery and structures can now be loaded as well eg weapon charging racks). 
tweak: Vehicle code has been partially rewritten and cargo or vehicle loading will work slightly differently. Report bugs on the github
tweak: Nerfs needlers so they require 8 shards to supercombine and have no armour penetration down from 20
bugfix: Significantly optimises packwar by preventing geysers from triggering lighting updates. There might be some improvements to other gamemodes
/:cl:

### CODE REVIEW

**Comment by BDpuffy420** [comment]
Can't be merged until these merge conflicts are sorted

**Comment by BDpuffy420** [comment]
Merge conflict sorted but Travis be angry

File: code/modules/halo/vehicles/vehicle_cargo.dm [hunk 1]
```diff
@@ -0,0 +1,148 @@
+
+/obj/vehicles/proc/can_put_cargo(var/obj/object)
+	world << "/obj/vehicles/proc/can_put_cargo([object] [object.type])"
```
Comment (bloxgate) [inline code/modules/halo/vehicles/vehicle_cargo.dm]: `<<` -> `to_world()`

Comment (bloxgate) [reply code/modules/halo/vehicles/vehicle_cargo.dm]: Travis is also complaining about there being a `/obj/` path here as text.

**Review by bloxgate — DISMISSED** [review]
This should fix the Travis code test.

The dosaic problem is being caused by a bad image path somewhere.

**Comment by caelaislinn** [comment]
Passing travis except for Achlys failure

**Review by BDpuffy420 — APPROVED** [review]

**Pull request merged**